File: python/FileUtils.py
from NodeFromFile import NodeFromFile


def read_nodes_from_file(file_name):
    nodes_from_file_list = []

    with open(file_name) as lines:
        for line in lines:
            splitted_line = line.rstrip().split(',')
            nodes_from_file_list.append(
                NodeFromFile(splitted_line[0], splitted_line[1], splitted_line[2], splitted_line[3]))
    return nodes_from_file_list

File: c#/FileUtils.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace binaryTree
{
    class FileUtils
    {
        public static List<NodeFromFile> ReadNodesFromFile(string fileName)
        {
            List<NodeFromFile> nodesFromFileList = new List<NodeFromFile>();

            try
            {
                string[] allLines = File.ReadAllLines(fileName);
                foreach (var line in allLines)
                {
                    var nodeDetails = line.Split(',');
                    nodesFromFileList.Add(
                            new NodeFromFile(
                                 nodeDetails[0],
                                 nodeDetails[1],
                                 nodeDetails[2],
                                 nodeDetails[3]
                                ));
                }
            } catch(Exception e)
            {
                Console.WriteLine(e.Message);
            }

            return nodesFromFileList;
        }
    }
}
